### PR TITLE
Check package properties in gremlin

### PR DESCRIPTION
**Pull request by tisnik**

### CODE REVIEW

File: integration-tests/features/steps/gremlin.py [hunk 1]
```diff
@@ -28,18 +29,25 @@ def gremlin_search_vertexes_for_the_ecosystem(context, ecosystem):
     post_query(context, query)
 
 
-@when('I ask Gremlin to find the package {package:S} in the ecosystem {ecosystem}')
+@when('I ask Gremlin to find all versions of the package {package:S} in the ecosystem {ecosystem}')
 def gremlin_find_package(context, package, ecosystem):
-    """Try to find the package in the selected ecosystem."""
-    query = Query().has("pecosystem", ecosystem).has("ppackage", package)
+    """Try to find all versions of the given package in the selected ecosystem."""
+    query = Query().has("pecosystem", ecosystem).has("pname", package)
```
Comment (miteshvp) [inline integration-tests/features/steps/gremlin.py]: If you really want to make use of indexes, you can just use 
`Query().has("pname", package)` which will still give you all the version nodes given this pname

File: integration-tests/features/steps/gremlin.py [hunk 2]
```diff
@@ -28,18 +29,25 @@ def gremlin_search_vertexes_for_the_ecosystem(context, ecosystem):
...
     """Try to find the package with version in the selected ecosystem."""
-    query = Query().has("pecosystem", ecosystem).has("ppackage", package).has("version", version)
+    query = Query().has("pecosystem", ecosystem).has("pname", package).has("version", version)
+    post_query(context, query)
+
+
+@when('I ask Gremlin to find the package {package:S} in the ecosystem {ecosystem}')
+def gremlin_find_package(context, package, ecosystem):
+    """Try to find the package in the selected ecosystem."""
+    query = Query().has("ecosystem", ecosystem).has("name", package).has("vertex_label", "Package")
```
Comment (miteshvp) [inline integration-tests/features/steps/gremlin.py]: Again, no need to provide vertex_label here. Just 
`query = Query().has("ecosystem", ecosystem).has("name", package)` should suffice

**Comment by miteshvp** [comment]
Good one. Just some minor nitpicks to make it more performant.

File: integration-tests/features/steps/gremlin.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 from behave import given, then, when
 from urllib.parse import urljoin
 from src.json_utils import *
+from src.utils import *
```
[the remark this answers is not in the document]
Comment (tisnik) [reply integration-tests/features/steps/gremlin.py]: actually just one function needs to be imported. going to fix it

File: integration-tests/features/steps/gremlin.py (hunk 2 above)
Comment (tisnik) [reply integration-tests/features/steps/gremlin.py]: good to know, thanks @miteshvp

File: integration-tests/features/steps/gremlin.py (hunk 1 above)
Comment (tisnik) [reply integration-tests/features/steps/gremlin.py]: How it works for packages with the same name, but created for different ecosystems?

File: integration-tests/features/steps/gremlin.py (hunk 2 above)
Comment (tisnik) [reply integration-tests/features/steps/gremlin.py]: fixed this one, thanks

File: integration-tests/features/steps/gremlin.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (tisnik) [reply integration-tests/features/steps/gremlin.py]: fixed this one, thanks

File: integration-tests/features/steps/gremlin.py (hunk 1 above)
Comment (miteshvp) [reply integration-tests/features/steps/gremlin.py]: @tisnik - that's a good question. If I may ask, what is the requirement here? You need to fetch all the versions for a given ecosystem + package?

Comment (tisnik) [reply integration-tests/features/steps/gremlin.py]: yes

Comment (miteshvp) [reply integration-tests/features/steps/gremlin.py]: ok. In that case, you should use something like
```
For picking only version numbers
g.V().has('ecosystem','maven').has('name','pkg-name').out('has_version').valueMap('version')

or for getting entire data for a version node
g.V().has('ecosystem','maven').has('name','pkg-name').out('has_version')
```
This query depending on the size of the data can be 10x more responsive due to proper usage of indexes. Hope it helps.

Comment (tisnik) [reply integration-tests/features/steps/gremlin.py]: Indeed, it helps me a lot. Thank you @miteshvp, I'll update the PR

**Comment by tisnik** [comment]
WDYT now guys?

**Pull request merged**